### PR TITLE
Created note listing page UI

### LINKED ISSUE #1: Create note listing page UI

### PR DESCRIPTION
**Pull request by adil62**
Fixes #1

### CODE REVIEW

File: app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import React from "react";
+
+import { MenuHorizontal, MenuVertical } from "neetoicons";
+import { Dropdown } from "neetoui";
+
+const EditOrDeleteActionDropdown = ({
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx:6]: This name is good. But we can keep it more compact and on point:
```suggestion
const ActionDropdown = ({
```

^ The above name is pretty self-explanatory and conveys what the other name was trying to point out.

File: app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  entityId,
+  onEditClick,
+  onDeleteClick,
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx:9]: These are handlers. So prefix them with the `handle` keyword. We only need to prefix with the `on` keyword, if that function emits an event.

File: app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  const listItems = [
+    { name: "Edit", clickHandler: () => onEditClick(entityId) },
+    { name: "Delete", clickHandler: () => onDeleteClick(entityId) },
+  ];
+  const dropdownIcon = isVerticalIcon ? MenuVertical : MenuHorizontal;
+
+  return (
+    <Dropdown buttonStyle="text" icon={dropdownIcon} position="bottom-start">
+      {listItems.map(({ name, handler }) => (
+        <li key={name} onClick={handler}>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx:21]: Had you done a thorough self-review from your end before merging this piece of code? What is `handler` in this scope?

Lack of self-review is not acceptable in BB. Please thoroughly go through the following docs: 
- https://handbook.neetokb.com/articles/git-and-github-essentials#self-review-process
- https://handbook.neetokb.com/articles/day-2#articles-to-be-read

File: app/javascript/src/components/Dashboard/Notes/Card.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        entityId={id}
+        onDeleteClick={onDeleteActionClick}
+        onEditClick={onEditActionClick}
+      />
+    </div>
+    <div>
+      <Typography style="h5" weight="normal">
+        {description}
+      </Typography>
+    </div>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Card.jsx:30]: Why we we need a `div` here?

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 5]
```diff
@@ -27,76 +28,83 @@ const Notes = () => {
   const fetchNotes = async () => {
     try {
       setLoading(true);
-      const { data } = await notesApi.fetch();
-      setNotes(data.notes);
+      const { data } = await notesApi.index();
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:31]: That's the wrong API connector name. We use `fetch` in front-end side to fetch something from the `index` action in backend.

Read: https://www.bigbinary.com/books/learn-rubyonrails-book/building-and-organizing-apis

File: app/javascript/src/utils/index.js [hunk 6]
```diff
@@ -1,3 +1,6 @@
+import dayjs from "dayjs";
 import * as R from "ramda";
 
 export const isPresent = R.pipe(R.either(R.isNil, R.isEmpty), R.not);
+
+export const calculateCreatedAgo = date => dayjs(date).fromNow();
```
Comment (yedhink) [inline app/javascript/src/utils/index.js:6]: Why do we need to keep this util function in the global `src/utils`? Are we using it at more than one place right now? If not, move this to an `utils.js` within the closest scope.

File: package.json [hunk 7]
```diff
@@ -60,6 +60,7 @@
   },
   "dependencies": {
     "@babel/preset-react": "^7.16.7",
+    "@bigbinary/neeto-commons-frontend": "^2.0.7",
```
Comment (yedhink) [inline package.json:63]: Why do we need this package for this challenge?

**Review by yedhink — COMMENTED** [review]
@adil62

File: app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx (hunk 3 above)
Comment (adil62) [reply app/javascript/src/components/Dashboard/EditOrDeleteActionDropdown.jsx:21]: **handler** should have been **clickHandler**  in that scope.
during the self-review process I noticed that **handler** was not a good name for a function so I renamed it to **clickHandler** in the listItems variable, but forgot to change it in the **li** element & committed.
Will be more careful next time, thanks.

File: package.json (hunk 7 above)
Comment (adil62) [reply package.json:63]: I thought id use the dynamicArray function from neeto-commons to generate the LIST_NOTES_DUMMY_DATA.
This is unneeded.
So I will remove this package in the next PR.

**Pull request merged**